### PR TITLE
doc: support short name S for state/province

### LINKED ISSUE #287: Support x509 subject with RDN state/province using short name "S"
The current [trust policy and store specification](https://github.com/notaryproject/specifications/blob/main/specs/trust-store-trust-policy.md#trusted-identities-constraints) has constraints on x509 subject DN. The state or province RDN need to be `ST=xxx`, for example `ST=Washington`, however `S=` is also an acceptable short name. If users set x509 subject DN with `S=xxx`, the trust policy will fail with the following error:

```
Error: trust policy statement "mypolicy" has trusted identity "x509.subject: xxxxxx" with invalid identity value: distinguished name (DN) "xxxxxx" has no mandatory RDN attribute for "ST", it must contain 'C', 'ST', and 'O' RDN attributes at a minimum
```

This issue is to ask for the support of state/province RDN with short name `S`.

### PR DESCRIPTION
**Pull request by yizha1**
Resolve #287 

Signed-off-by: Yi Zha <yizha1@microsoft.com>

### CODE REVIEW

**Comment by yizha1** [comment]
@priteshbandi @shizhMSFT @Two-Hearts @JeyJeyGao Would you mind taking a look at this PR?

**Review by ghost — APPROVED** [review]
LGTM

**Review by shizhMSFT — COMMENTED** [review]
Will this PR cover #293 and #298?

**Comment by yizha1** [comment]
@shizhMSFT no, only fixing 287 to support additional `S`.

**Review by shizhMSFT — APPROVED** [review]
LGTM. It is better to have a reference where `S` and `ST` come from.

File: specs/trust-store-trust-policy.md [hunk 1]
```diff
@@ -442,9 +442,9 @@ The RDN consists of an attribute type name followed by an equal sign and the str
   - The value of each `trustedIdentities` list item, if it begins with `x509.subject:`, MUST be followed by comma-separated one or more RDNs.
     Other types of trusted identities may be supported, by using an alternate prefix, or a different format.
     For example, `x509.subject: C=${country}, ST=${state}, L=${locallity}, O={organization}, OU=${organization-unit}, CN=${common-name}`.
-  - Each identity in `identities` list MUST contain country (C), state or province (ST), and organization (O) RDNs.
+  - Each identity in `identities` list MUST contain country (C), state or province (ST or S), and organization (O) RDNs.
     All other RDNs are optional.
-    The minimal possible value is `x509.subject: C=${country}, ST=${state}, O={organization}`,
+    The minimal possible value is `x509.subject: C=${country}, ST=${state}, O={organization}` or `x509.subject: C=${country}, S=${state}, O={organization}`.
```
Comment (priteshbandi) [inline specs/trust-store-trust-policy.md:447]: Shouldn't it be` C=${country}, ST=${state}, O={organization}` Or `C=${country},  L=${localityName}, O={organization}` ?

Comment (yizha1) [reply specs/trust-store-trust-policy.md:447]: This PR is only for the short name `S` that can be used for province/state as well, currently only `ST` is allowed, but `S` is also correct. Using `L` is a different issue, which is tracked here https://github.com/notaryproject/specifications/issues/293, but there is no decision yet.

**Review by toddysm — APPROVED** [review]

File: specs/trust-store-trust-policy.md (hunk 1 above)
Comment (priteshbandi) [reply specs/trust-store-trust-policy.md:447]: Can you share RFC which defines `S` and is oid for `S` different from [`ST`](http://www.oid-info.com/get/2.5.4.8) ?

Comment (yizha1) [reply specs/trust-store-trust-policy.md:447]: @priteshbandi the oid is the same. In the section "6.3.3" of [ITU-X.520](https://www.itu.int/rec/dologin_pub.asp?lang=e&id=T-REC-X.520-201210-S!!PDF-E&type=items), `S` is used in the example, although `ST` is used for the LDAP name. Another doc also suggests `S` not `ST`, see https://learn.microsoft.com/en-us/windows/win32/api/wincrypt/nf-wincrypt-certnametostra?redirectedfrom=MSDN#remarks. So, there are some cases that `S` is used. Allowing the use of either 'S' or 'ST' as the field label for State/Province will be beneficial.

**Review by priteshbandi — APPROVED** [review]
LGTM

**Comment by priteshbandi** [comment]
Can you please open an issue in notation-go to implement this?

**Pull request merged**